scripts/migrateImageRoles.py

```diff
@@ -0,0 +1,80 @@
+import re
+
+from bson.objectid import ObjectId
+from girder.utility import server as girder_server  # noqa
+
+girder_server.configureServer()  # noqa
+from girder.models.file import File  # noqa
+from girder.models.item import Item  # noqa
+
+from isic_archive.models.dataset import Dataset  # noqa: E402
+
+
+for image in Item().find(
+    {'$match': {'baseParentId': ObjectId('55943cff9fc3c13155bcad5e')}}
+):
+    dataset = Dataset().find({'_id': image['meta']['datasetId']})
+    print(f'{image["name"]}, {image["_id"]}, {next(dataset)["name"]}')
+    files = list(File().find({'itemId': image['_id']}))
+    taken_file_ids = set()
+
+    superpixel_file_ids = [
+        f['_id']
+        for f in files
+        if 'superpixels' in f['name']
+        and f['size'] > 0
+        and f['_id'] not in taken_file_ids
+    ]
+    taken_file_ids = taken_file_ids.union(set(superpixel_file_ids))
+
+    exif_file_ids = [
+        f['_id']
+        for f in files
+        if '.stripped.' in f['name']
+        and f['size'] > 0
+        and f['_id'] not in taken_file_ids
+    ]
+    taken_file_ids = taken_file_ids.union(set(exif_file_ids))
+
+    large_image_file_id = (
+        image['largeImage']['fileId'] if 'largeImage' in image else None
+    )
+    if large_image_file_id:
+        assert large_image_file_id not in taken_file_ids
+        taken_file_ids = taken_file_ids.union(set([large_image_file_id]))
+
+    original_files = [
+        f['_id']
+        for f in files
+        if re.match('^ISIC_\d{7}\.[^\.]+$', f['name'])  # noqa
+        and f['_id'] not in taken_file_ids
+    ]
+
+    # this happens with the VSHR dataset
+    if len(original_files) > 1:
+        print('warning: more than one original file, picking at random')
+    original_file_id = original_files[0]
```

Comment:
Are all of them identical? Is the 0th arbitrary or always the correct one?

Reply:
I resolved this to fall back to the jpeg if they aren't identical. The only place this occurred was in a dataset with duplicate large images.